### ISSUE TITLE
UnicodeDecodeError while parsing

### ISSUE THREAD

**Issue by joszko** [fredrikaverpil/creosote#124]
Hi, 

I'm using creosote 2.2.0 and while parsing I get a following error:

> Traceback (most recent call last):
>   File "C:\Users\8008701\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "C:\Users\8008701\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
>     exec(code, run_globals)
>   File "C:\Users\8008701\PycharmProjects\onesource_idt-intl-tools\venv_test\Scripts\creosote.exe\__main__.py", line 7, in <module>
>   File "C:\Users\8008701\PycharmProjects\onesource_idt-intl-tools\venv_test\lib\site-packages\creosote\cli.py", line 77, in main
>     imports = parsers.get_modules_from_code(args.paths)
>   File "C:\Users\8008701\PycharmProjects\onesource_idt-intl-tools\venv_test\lib\site-packages\creosote\parsers.py", line 163, in get_modules_from_code
>     for imp in get_module_info_from_code(resolved_path):
>   File "C:\Users\8008701\PycharmProjects\onesource_idt-intl-tools\venv_test\lib\site-packages\creosote\parsers.py", line 137, in get_module_info_from_code
>     root = ast.parse(fh.read(), path)
>   File "C:\Users\8008701\AppData\Local\Programs\Python\Python310\lib\encodings\cp1252.py", line 23, in decode
>     return codecs.charmap_decode(input,self.errors,decoding_table)[0]
> UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 3560: character maps to <undefined>

All files are in UTF-8

**Comment by fredrikaverpil** [comment]
Hi @joszko and thanks for reporting!

Hm, this was a weird one 😅. Let me dig a bit. I'm not sure how to reproduce this one.

**Comment by BrianPugh** [comment]
@fredrikaverpil you might need to add `encoding='utf-8'` to all your open calls.

Alternatively, @joszko try setting the envvar `PYTHONUTF8=1` and see if that fixes anything.

**Comment by joszko** [comment]
@BrianPugh adding envar fixed it, all files were parsed. Thanks!

**Comment by BrianPugh** [comment]
cool! so adding `encoding='utf-8'` to all open calls should probably solve this.

**Comment by fredrikaverpil** [comment]
Oh man, I don't think I've seen encoding issues since the python 2.7 days... how come this happens, do you know @BrianPugh ?

I've opened #136 but I'm a little worried this might cause issues on some systems.

**Comment by fredrikaverpil** [comment]
@joszko if you can confirm that this works too on your end, that would be great 😄 

```bash
# install in-development version from PR #136
pipx install --suffix=@136 --force git+https://github.com/fredrikaverpil/creosote.git@refs/pull/129/head

# run creosote in-development version
creosote@136 -v .venv ... --verbose

# uninstall in-development version
pipx uninstall creosote@136
```

**Comment by BrianPugh** [comment]
All I know about encoding is from this recent mcoding video (jump to minute 5) 😄 :

https://www.youtube.com/watch?v=EimoZHDcQMA

**Comment by fredrikaverpil** [comment]
@joszko when I'm reading about [`open`](https://docs.python.org/3/library/functions.html#open), this caught my eye:

> if encoding is not specified the encoding used is platform-dependent: [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding)

What do you get if you run this?

```bash
import locale
print(locale.genencoding())
```

**Comment by jonded94** [comment]
@fredrikaverpil 
Just one click further could have given you the answer :) https://docs.python.org/3/library/locale.html#locale.getencoding

On Linux and MacOS, `open` will almost always default to UTF-8, unless specifically set by your distro or user to something else. On Windows the default is something like cp1252, cp932, etc. @joszko clearly seems to be on Windows.

There is an open PEP that will set the default to UTF-8 across the board but will not be implemented earlier than Python 3.15

https://peps.python.org/pep-0686/#:~:text=The%20default%20encoding%20of%20Python,use%20UTF%2D8%20by%20default.
